text/0911-const-fn.md

```diff
@@ -172,6 +172,18 @@ const fn arithmetic_ops<T: Int>() -> [fn(T, T) -> T; 4] {
 }
 ```
 
+`const` functions can also be unsafe, allowing construction of types that require
+invariants to be maintained (e.g. `std::ptr::Unique` requires a non-null pointer)
+```rust
+struct OptionalInt(u32);
+impl OptionalInt {
+    /// Value must be non-zero
+    unsafe const fn new(val: u32) -> OptionalInt {
```

Comment:
@eddyb I see how a const fn ptr could allow for higher order functions at compile-time, I am not sure that is the right means to achieve that.